### ISSUE TITLE
Invalid Hook Call, inside react-native-picker-select

### ISSUE THREAD

**Issue by emilseuor** [lawnstarter/react-native-picker-select#486]
Invalid Hook Call error inside react-native-picker-select node module.

![image](https://user-images.githubusercontent.com/13121388/193236994-35da61fa-e583-4f06-8e94-378558ec65ed.png)

**Describe the bug**<br />
I get this issue everytime I build in android, it would show "Invalid Hook Call error" when it has to render the screen that uses RNPickerSelect (react-native-picker-select)

Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.

This error is located at:
    in PickerAndroid (at Picker.js:174)
    in Picker (at src/index.js:491)
    in RCTView (at View.js:32)
    in View (at src/index.js:489)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:211)
    in AnimatedComponent (at createAnimatedComponent.js:264)
    in AnimatedComponentWrapper (at TouchableOpacity.js:224)
    in TouchableOpacity (at TouchableOpacity.js:280)
    in TouchableOpacity (at src/index.js:483)
    in RNPickerSelect (at Picker.tsx:39)
    in RCTView (at View.js:32)
    in View (at Picker.tsx:38)
    in RCTView (at View.js:32)
    in View (at Picker.tsx:36)

**To Reproduce**<br />

1.  Add react-native-picker-select on a react native app (react-native init Test)
2.  Add RNPickerSelect useNativeAndroidPickerStyle={false} in a screen with data
3.  Build in android (react-native run-android)
4.  See error

**Expected behavior**<br />
It would render the picker and not crash the app

**Screenshots**<br />
![image](https://user-images.githubusercontent.com/13121388/193238236-161078e1-2bc1-4d16-b229-31657444bdff.png)
![image](https://user-images.githubusercontent.com/13121388/193238292-d13bc7c7-21cb-4ab6-8968-54bb4fae5e6b.png)

**Additional details**<br />
-   Device: Android 11G Emulator
-   OS: Android 11
-   react-native-picker-select version: ^8.0.4
-   react-native version: ^0.68.5
-   expo sdk version: n/a

**Reproduction and/or code sample**<br />

<RNPickerSelect
  placeholder={{label: 'test', value: 'test'}}
  items={items}
  onValueChange={props.functionOnChange}
  style={styles}
  useNativeAndroidPickerStyle={false}
/>

**Comment by seb-glanum** [comment]
I have the same issue :(

**Comment by seb-glanum** [comment]
Okay I found a workaround to fix it. 
In my case I was using npm and there is an issue with peer dependencies and React 18 used on newer versions of React Native.

I have re-install the module with this command  => `npm i react-native-picker-select --legacy-peer-deps`

**Comment by digitalbore** [comment]
same problem here

**Comment by Nestoro** [comment]
same problem on react-native 0.69.1 and react 18.0.0

**Comment by emilseuor** [comment]
I solved it with npm i --legacy-peer-deps 

And also making sure that the data I send on items is not undefined.

**Comment by kueda** [comment]
FWIW, the issue seems to be with `@react-native-picker/picker`: https://github.com/react-native-picker/picker/issues/424

**Comment by chinmayk3** [comment]
yes . npm i react-native-picker-select --legacy-peer-deps worked for me . Thanks @seb-glanum

**Comment by khalidMhd** [comment]
> Okay I found a workaround to fix it. In my case I was using npm and there is an issue with peer dependencies and React 18 used on newer versions of React Native.
> 
> I have re-install the module with this command => `npm i react-native-picker-select --legacy-peer-deps`

but it still crashes on the build android APK. WHY?

**Comment by ollieyeahlong** [comment]
Still crashes on Andoid APK for me too.

**Comment by nlarif** [comment]
Solved for me.
I faced the same issue. I found out that react-native-picker-select was installing on old version of @react-native-picker/picker under node_modules/react-native-picker-select/node_modules/@react-native-picker/picker which was the one throwing the exception.
I updated my packages.json with:

```
"resolutions": {
   "react-native-picker-select/@react-native-picker/picker": "^2.4.8"
 },
```

and used Yarn locally and in my CI/CD. I removed the usage of `npm install --save --legacy-peer-deps` which is no longer needed.
Now react-native-picker-select is relying on the newest version within node_modules and I don't see the error anymore.

**Comment by ErAmanDhiman** [comment]
You only need to add 
<application android:supportsRtl="true">
in AndroidManifest.xml

**Comment by francosbenitez** [comment]
Not working.

**Comment by ipakhomov** [comment]
For those who use solution with `--legacy-peer-deps` flag: if your app works in Expo Go, but crashes in standalone build created with EAS Build, you may to add the following script in the `"scripts"` section of `package.json`:
```
    "eas-build-pre-install": "npm config set legacy-peer-deps true",
```

cc @khalidMhd @ollieyeahlong @francosbenitez

**Comment by OceanDeverloper** [comment]
> Okay I found a workaround to fix it. In my case I was using npm and there is an issue with peer dependencies and React 18 used on newer versions of React Native.
> 
> I have re-install the module with this command => `npm i react-native-picker-select --legacy-peer-deps`

Thanks so much!

**Comment by stef1103** [comment]
I found another solution which is adding this to the package json:

`"overrides": {
    "react-native-picker-select": {
      "@react-native-picker/picker": "$@react-native-picker/picker"
    }
  }`

For npm this override the package dependency forcing it to use the same version set in the "main" dependencies.
Doing this react-native-picker-select will not use the old @react-native-picker/picker

**Comment by rnnyrk** [comment]
@stef1103 Thank a lot. Works perfect!

**Comment by cavemon** [comment]
@stef1103 Thank you!  

FWIW, I had to delete my `react-native-picker-select` folder from `node_modules`, then run `npm i` again.

**Comment by kubacienciala** [comment]
@cavemon do i need to install react-native-picker/picker also?

**Comment by cavemon** [comment]
> @cavemon do i need to install react-native-picker/picker also?

Yes. For clarity sake, here is a snippet of my `package.json`

```json
"dependencies": {
    "react-native-picker-select": "^8.0.4",
},
"overrides": {
    "react-native-picker-select": {
      "@react-native-picker/picker": "$@react-native-picker/picker"
    }
  }
  ```

**Comment by jimmy-qian** [comment]
This solution works great

> I found another solution which is adding this to the package json:
> 
> ```
> "overrides": {
>     "react-native-picker-select": {
>       "@react-native-picker/picker": "$@react-native-picker/picker"
>     }
>   }
> ```
> 
> For npm this override the package dependency forcing it to use the same version set in the "main" dependencies. Doing this react-native-picker-select will not use the old @react-native-picker/picker

This is the way to go for now.

**Comment by mahmoudkandel980** [comment]
i tried a lot of solutions in above but some of them work for me but never working again after closing ide. i found in there documentation in GitHub that you should run one more command. i didn't saw it because i install packages from npm page.

# React Native users
npm install react-native-picker-select
npm install @react-native-picker/picker
npx pod-install

# Expo
npm install react-native-picker-select
expo install @react-native-picker/picker

**Comment by furkanuzun** [comment]
Same issue here.
